src/main/java/calculator/view/InputView.java

```diff
@@ -0,0 +1,48 @@
+package calculator.view;
+
+import calculator.domain.Operand;
+import calculator.domain.Operator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class InputView {
+    private final static String SEPARATOR = " ";
+    private final String formula;
+    private final List<Operand> operands = new ArrayList<>();
+    private final List<Operator> operators = new ArrayList<>();
+
+    public InputView(String formula) {
+        if (formula == null || formula.isBlank()) {
+            throw new IllegalArgumentException("null 또는 빈 공백 문자를 전달할 수 없습니다.");
+        }
+
+        Arrays.stream(formula.split(SEPARATOR))
+                .forEach(s -> {
+                    if (isOperand(s)) {
+                        operands.add(new Operand(Double.valueOf(s)));
+                    } else {
+                        operators.add(new Operator(s));
+                    }
+                });
+        this.formula = formula;
+    }
```

Comment:
```suggestion
    public List<Operand> getOperands(String formula) {
        if (formula == null || formula.isBlank()) {
            throw new IllegalArgumentException("null 또는 빈 공백 문자를 전달할 수 없습니다.");
        }

        return Arrays.stream(formula.split(SEPARATOR))
                .filter(this::isOperand)
                .map(Operand::new)
                .collect(Collectors.toList());
    }

    public List<Operator> getOperators(String formula) {
        if (formula == null || formula.isBlank()) {
            throw new IllegalArgumentException("null 또는 빈 공백 문자를 전달할 수 없습니다.");
        }
        
        return Arrays.stream(formula.split(SEPARATOR))
                .filter(value -> !isOperand(value))
                .map(Operator::new)
                .collect(Collectors.toList());
    }
```
- 위와 같이 각 기능별로 메소드를 나누는건 어떨까요? 위 구성은 장단점이 있기는 합니다.
  - 반복문을 두 번 수행한다는 단점이 존재
  - 기능 단위로 테스트 가능하고 가독성이 높음
  - 뷰모델 생성시 모든 객체 생성 및 유효성을 검증하지 않고, getOperands, getOperators 를 호출하는 시점에 로직을 수행함

Reply:
메소드를 나누어 볼 생각을 못했는데 좋은 방법 알려주셔서 감사합니다!
분리하니 제가 다시 보기에도 수월해졌어요 ! 👍